src/controllers/components/omnibar.js

```diff
@@ -32,6 +32,23 @@ export default class OmnibarController {
             if (StudentStore.getStudent() !== null) {
                 currentStudentId = StudentStore.getStudent().id;
             }
+            if (CartStore.getIsLongterm()) {
+                if (!StudentController.isValidLongtermData(CartStore.getDueDate(), CartStore.getProfessor())) {
+                    return;
+                } else {
+                    return StudentController.longtermCheckout(StudentStore.getStudent().id, CartStore.getContents(), CartStore.getDueDate(),
+                        CartStore.getProfessor()).then(() => {
+                            if (CartStore.getContents().length === 0) {
+                                return searchStudent(id);
+                            }
+                        }).then(student => {
+                            if (CartStore.getContents().length === 0) {
+                                Dispatcher.handleAction("STUDENT_FOUND", student);
+                                hashHistory.push('/student');
+                            }
+                        });
+                }
+            }
             return StudentController.checkout(currentStudentId, CartStore.getContents()).then(() => {
```

Comment:
Shouldn't there be an `else` here, so there isn't an implicit stop-executing-this-function at the `hashHistory.push` above&hellip;